### ISSUE TITLE
better type hint for expected condition?

### ISSUE THREAD

**Issue by bandophahita** [bandophahita/screenpy#9]
This feels like a really clean type hint,
```python
@runtime_checkable
class ExpectedCondition(Protocol):
    def __init__(self, *args): pass
    def __call__(self, driver: "WebDriver"): pass
```

The problem is it requires importing from different libraries depending on the python version.  I don't have much experience in ensuring dynamic imports for a package.  Perhaps there is a better way to manage this? Maybe a try/except would be better?
```python
import sys
if sys.version_info < (3, 8, 0):
    from typing_extensions import Protocol, runtime_checkable  # after python3.8 this can be imported from typing directly
else:
    from typing import Protocol, runtime_checkable
```

**Comment by perrygoy** [comment]
`Protocol` didn't leave `typing_extensions` in 3.9, so we can just keep importing it from `typing_extensions` for now. I plan on moving over to the `typing` import probably along with Python 4 compatibility updates, since that feels like a clean time to do it.

**Comment by bandophahita** [comment]
That makes sense.  

I'm curious your thoughts about this type of type hint?  Is it overkill?

**Comment by bandophahita** [comment]
The `ExpectedCondition` type hint.

**Comment by perrygoy** [comment]
Yes, i'm asking where this protocol will be used. What will need to follow it? I'm guessing based on the name and the signature that it's probably for Resolutions, right?

**Comment by bandophahita** [comment]
> (Also, sorry to barge in on your fork, i accidentally got here and didn't notice at first.)

No problem at all.  In fact I appreciate it.

I was thinking of `Wait.using`.  I thought the rules for callabales that are passed into WebDriverWait are required accept the driver even if they don't utilize it.  
```python
    def using(
        self, strategy: ExpectedCondition, log_detail: Optional[str] = None
    ) -> "Wait":
```
I like pycharm a lot and I really like it when it type hints can point out mistakes.  I've found that when arguments are Callable it's easy to miss less obvious problems like first argument isn't a `WebDriver`.  

What I'm not sure about is if using this type of hint is overkill just to get an IDE to point out mistakes.

**Comment by perrygoy** [comment]
Oooh Selenium's `ExpectedCondition`, i'm following now. Sorry, i'm still waking up apparently. :P

Maybe this is actually pointing to a problem in ScreenPy, where there would be a better type hint from Selenium to be using. Otherwise, it might be useful to spell out the protocol, yeah.

**Comment by bandophahita** [comment]
The current type hint isn't wrong by any measure, so I hesitate to call it a problem.  `Callable[..., Any]`

You make a good point about using a type hint from selenium.  Last I recall selenium 2 library didn't have a good type hint; in fact I don't think they included ANY type hints in that entire library.  I haven't moved up to selenium 3 yet but think maybe it's time to start thinking about doing so.
